block/manager.go

```diff
@@ -317,7 +317,28 @@ func NewManager(
 		return nil, err
 	}
 
-	if s.DAHeight < config.DA.StartHeight {
+	// Determine the DA height to start from based on the last applied block's DA inclusion
+	if s.LastBlockHeight > 0 {
+		// Try to find where the last applied block was included in DA
+		headerKey := fmt.Sprintf("%s/%d/h", storepkg.HeightToDAHeightKey, s.LastBlockHeight)
+		if daHeightBytes, err := store.GetMetadata(ctx, headerKey); err == nil && len(daHeightBytes) == 8 {
+			lastBlockDAHeight := binary.LittleEndian.Uint64(daHeightBytes)
+			// Start scanning from the next DA height after the last included block
+			s.DAHeight = lastBlockDAHeight + 1
+			logger.Info().
+				Uint64("lastBlockHeight", s.LastBlockHeight).
+				Uint64("lastBlockDAHeight", lastBlockDAHeight).
+				Uint64("startingDAHeight", s.DAHeight).
+				Msg("resuming DA scan from last applied block's DA inclusion height")
+		} else {
+			// Fallback: if we can't find DA inclusion info, use the persisted DA height
+			logger.Info().
+				Uint64("lastBlockHeight", s.LastBlockHeight).
+				Uint64("daHeight", s.DAHeight).
```

Comment:
Is the DA height actually persisted? If it was then why would it always start from 0?

Reply:
i think there is another bug, that is hidden here. ive been going through the code slowly. there is a startup discrepancy